Monika-After-Story/MonikaModDev
# Generalized Dockstat Flow

Adjusted the dockstat flow such that it's completely modular. Simply passing in a label for a specific part will replace that part of the generic flow with the one provided (should the label exist)

## Added:
- 6 new variables to control the dockstat flow
  - `dockstat_iowait_label`:
     - The label used for the "let me get ready" phase of the io generation (Used by the iostart label)

  - `dockstat_rtg_label`:
     - The label containing dialogue for when Monika is ready to go out (Used by the generic iowait label)

  - `dockstat_cancel_dlg_label`:
     - The label containing dialogue used for when you tell Monika you can't take her with you (Used in the generic iowait label)

  - `dockstat_wait_menu_label`:
    - The label used to contain the menu in which Monika asks what's wrong (Used when you click the wait button during iowait)

  - `dockstat_cancelled_still_going_ask_label`:
    - The label used which contains the menu where Monika asks the player if they're still going to go (Used if you cancel dockstat gen)

  - `dockstat_failed_io_still_going_ask_label`:
    - The label used which contains the menu where Monika asks the player if they're still going to go (Used in the generic rtg label if we failed to generate a file)

- If these are None or non-existent labels, the generic is used (generic being the current standard flow for the taking you somewhere farewell)

## Testing:
- Verify that the taking you somewhere farewell still works without issues
- Verify that the player bday going somewhere still works without issues
- Verify that O31 Trick or Treat fare still works without issues
- Verify that d25 dockstat still works without issues
- Verify that all vars are cleared on dockstat cancel
- Verify that plugging your own labels into the parts has them used accordingly
- Verify that the flow will always be generic if the label provided to be subbed in doesn't exist

### NOTE: This also comments out some CG code in Marisa/Rin o31 greets
